packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSourceCache.ts

```diff
@@ -3,56 +3,24 @@ import useLazyRef from '@mui/utils/useLazyRef';
 import { useGridApiMethod } from '@mui/x-data-grid';
 import { GridPrivateApiPro } from '../../../models/gridApiPro';
 import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
-import { GridGetRowsParams, GridGetRowsResponse, GridDataSourceCache } from '../../../models';
+import { GridDataSourceCache } from '../../../models';
 import { GridDataSourceCacheApi } from './interfaces';
 import { GridDataSourceDefaultCache } from './cache';
 
-const getDefaultCache = (cacheInstance: GridDataSourceDefaultCache): GridDataSourceCache => ({
-  getKey: cacheInstance.getKey,
-  set: (key: string, value: GridGetRowsResponse) =>
-    cacheInstance.set(key as string, value as GridGetRowsResponse),
-  get: (key: string) => cacheInstance.get(key as string),
-  clear: () => cacheInstance.clear(),
-});
-
 export const useGridDataSourceCache = (
   privateApiRef: React.MutableRefObject<GridPrivateApiPro>,
   props: Pick<DataGridProProcessedProps, 'unstable_dataSource' | 'unstable_dataSourceCache'>,
 ): void => {
-  const defaultCache = useLazyRef<GridDataSourceCache, void>(() =>
-    getDefaultCache(new GridDataSourceDefaultCache({})),
-  );
-  const cache = React.useRef<GridDataSourceCache | null>(
-    props.unstable_dataSourceCache || defaultCache.current,
-  );
+  const defaultCache = useLazyRef<GridDataSourceCache, void>(
+    () => new GridDataSourceDefaultCache({}),
+  ).current;
 
-  const getCacheData = React.useCallback((params: GridGetRowsParams) => {
-    if (!cache.current) {
-      return undefined;
-    }
-    const key = cache.current.getKey(params);
-    return cache.current.get(key);
-  }, []);
-
-  const setCacheData = React.useCallback((params: GridGetRowsParams, data: GridGetRowsResponse) => {
-    if (!cache.current) {
-      return;
-    }
-    const key = cache.current.getKey(params);
-    cache.current.set(key, data);
-  }, []);
-
-  const clearCache = React.useCallback(() => {
-    if (!cache.current) {
-      return;
-    }
-    cache.current.clear();
-  }, []);
+  const [cache, setCache] = React.useState<GridDataSourceCache | null>(
+    props.unstable_dataSourceCache !== undefined ? props.unstable_dataSourceCache : defaultCache,
+  );
```

Comment:
`what if props.unstable_dataSourceCache changes?`

It should be updated thanks to the `useGridApiMethod`, no?

> I feel like having two hooks for the cache here is a bit confusing.

Do you mean `useGridDataSource` and `useGridDataSourceCache`? If we see a clear problem here, we could probably remove `useGridDataSourceCache` and move the logic to `useGridDataSource`. Wdyt?

Reply:
I meant `useState` and `useLazyRef` but I also think it would be better to have everything in one file. If we remove (or have removed) the API cache methods, it should comfortably fit in one file.

Reply:
Ah, I get your point. Correct, we can initialize the default one in the `useState` initializer function.
I [moved](https://github.com/mui/mui-x/pull/12317/commits/655cc09cf072ec7057ae87760855709f0d40e108) it to the `useGridDataSource` hook and reduced it to a single hook.